### PR TITLE
CORE-1009 User can view community data without logging in

### PR DESCRIPTION
**Pull request by ash3rz**
NOTE: Requires [terrain #177](https://github.com/cyverse-de/terrain/pull/177) to be merged first.  Also needs config changes in k8s-resources

These are the Sonora changes to enable logged out users to browse community data, read public files, and search community data.  The breadcrumbs when logged out are slightly changed so that Community Data comes first, and what would have been the user's home folder is now generically named "Home".  Clicking on any breadcrumb other than Community Data will take the user to the "please sign in" page:

image

### CODE REVIEW

**Review by sri2k1us — APPROVED** [review]
LGTM 👍

File: src/constants.js [hunk 1]
```diff
@@ -61,4 +61,5 @@ export default {
         SPECIAL_CHARS:
             "https://cyverse.atlassian.net/wiki/spaces/DEmanual/pages/242027163/Troubleshooting+an+Analysis#TroubleshootinganAnalysis-3.Checktheinputfilesandparameter",
     },
+    ANONYMOUS_USER: "anonymous",
```
[the remark this answers is not in the document]
Comment (slr71) [reply src/constants.js:64]: I think that the `anonymous` and `public` accounts are always present in iRODS installations.

File: src/server/api/data.js [hunk 2]
```diff
@@ -119,7 +119,7 @@ export default function dataRouter() {
         auth.authnTokenMiddleware,
         terrainHandler({
             method: "GET",
-            pathname: "/secured/filetypes/type-list",
```
Comment (slr71) [inline src/server/api/data.js:122]: It's nice to see these `/secured` URL path prefixes going away. 👍

File: src/server/index.js [hunk 3]
```diff
@@ -84,7 +84,12 @@ app.prepare()
 
         logger.info("adding the /login/* handler");
         server.get("/login/*", keycloakClient.protect(), (req, res) => {
-            res.redirect(req.url.replace(/^\/login/, ""));
+            res.redirect(
+                req.url
+                    .replace(/^\/login/, "")
+                    .replace(`${config.iRodsHome}/anonymous`, "")
+                    .replace(`${config.iRodsTrash}/anonymous`, "")
+            );
```
[the remark this answers is not in the document]
Comment (slr71) [reply src/server/index.js:92]: It might be nice to have a comment explaining the reasoning in the code.

**Review by slr71 — APPROVED** [review]
LGTM! :shipit: whenever you're ready.

File: src/constants.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sri2k1us) [reply src/constants.js:64]: Gtk. In that please ignore my comments. I was under the impression we created those accounts.

[the remark this answers is not in the document]
Comment (slr71) [reply src/constants.js:64]: We actually kind of created the accounts and kind of didn't. The accounts are always present (or at least present by default) in iRODS, but we had to add them to LDAP to make things a little easier for ourselves.

**Comment by ash3rz** [comment]
The terrain PR has been merged and the config change has been pushed, so I'll merge this now.  Thanks for the reviews!

**Pull request merged**